spring-integration-core/src/main/java/org/springframework/integration/context/IntegrationObjectSupport.java

```diff
@@ -160,6 +175,13 @@ protected TaskScheduler getTaskScheduler() {
 		return this.taskScheduler;
 	}
 
+	protected DestinationResolver<MessageChannel> getChannelResolver() {
+		if (this.channelResolver == null) {
+			this.channelResolver = new BeanFactoryChannelResolver(this.beanFactory);
```

Comment:
Check for a `null` BF, in which case, perhaps default to a `FindNoneChannelResolver` to avoid possible NPEs ?